playbooks/ci-full/tasks/create.yml

```diff
@@ -48,11 +49,22 @@
         wait: true
       with_items: "{{ testenv_instance_names }}"
       register: testenv_floating_ips
+      when: use_private_ips == false
```

Comment:
You want to use the `|bool` filter here to help avoid comparing strings and breaking on falsey values like `no`.